Windows CLI fails to start daemon + headed mode ignored

## Summary

Two bugs affecting Windows users that prevent the agent-browser from working correctly:

1. **Headed mode is ignored** - The `AGENT_BROWSER_HEADED=1` environment variable has no effect
2. **Windows CLI fails with "Daemon failed to start"** - Path escaping issues when spawning the Node.js daemon

---

## Bug 1: Headed mode doesn't work (headless hardcoded)

### Location
`src/daemon.ts` (compiled as `dist/daemon.js`, around line 180)

### Problem
The daemon has `headless: true` hardcoded, completely ignoring the `AGENT_BROWSER_HEADED` environment variable:

```typescript
// Current code
const browser = await chromium.launch({
  headless: true,  // Always headless, ignoring env var
  // ...
});
```

### Expected Behavior
Setting `AGENT_BROWSER_HEADED=1` should launch the browser in headed mode for debugging.

### Fix
```typescript
const browser = await chromium.launch({
  headless: process.env.AGENT_BROWSER_HEADED !== '1',
  // ...
});
```

---

## Bug 2: Windows CLI fails with "Daemon failed to start"

### Location
`cli/src/connection.rs`

### Problem
On Windows, the Rust CLI spawns the Node.js daemon using `cmd /c`, which causes path escaping issues:

```rust
// Current approach (problematic on Windows)
Command::new("cmd")
    .arg("/c")
    .arg(&cmd_string)  // cmd_string = "node \"C:\path\to\daemon.js\""
```

This results in the path being mangled. The error message shows:

```
Cannot find module 'C:\Users\username\project\"C:\Users\username\AppData\Roaming\npm\node_modules\@anthropic-ai\agent-browser\dist\daemon.js"'
```

Notice how:
- The CWD gets prepended to the path
- The quotes become part of the filename instead of escaping the path

### Steps to Reproduce
1. Install agent-browser on Windows: `npm install -g @anthropic-ai/agent-browser`
2. Run any command that starts the daemon
3. Observe "Daemon failed to start" error

### Fix
Call Node.js directly instead of through `cmd.exe`:

```rust
// Instead of:
Command::new("cmd")
    .arg("/c")
    .arg(&cmd_string)

// Use:
Command::new("node")
    .arg(&daemon_path)
    .spawn()
```

This avoids the shell escaping issues entirely.

---

## Environment
- OS: Windows 10/11
- agent-browser version: latest from npm
- Node.js: v20+

## Impact
These bugs completely prevent Windows users from:
1. Using the agent-browser CLI at all (Bug 2)
2. Debugging browser automation in headed mode (Bug 1)

---

*Found while integrating agent-browser with Claude Code on Windows.*

--- a/src/daemon.ts
+++ b/src/daemon.ts
@@ -200,7 +200,7 @@ export async function startDaemon(options?: { streamPort?: number }): Promise<vo
             await browser.launch({
               id: 'auto',
               action: 'launch',
-              headless: true,
+              headless: process.env.AGENT_BROWSER_HEADED !== '1',
               executablePath: process.env.AGENT_BROWSER_EXECUTABLE_PATH,
               extensions: extensions,
             });

